File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution(object):
    def longestPalindrome(self, s):
        max_sub=s[0]
        for i in range(0,len(s)):
            for j in range(i+1,len(s)+1):
                cur_sub=s[i:j]
                rev_sub=cur_sub[::-1]
                if rev_sub==cur_sub and len(cur_sub)>len(max_sub):
                    max_sub=cur_sub             
        if (max_sub):
            return max_sub
        else:
            return ""

            
            
        

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
       if (s.isEmpty() || s.length()==0)
       {
        return "";
       }
       int start=0;int end=0;
       for (int i=0;i<s.length();i++)
       {
            int left=i;int right=i; //odd length
            while(left>=0 && right<s.length() && s.charAt(left)==s.charAt(right))
            {
                if ((right-left)>(end-start))
                {
                    start=left;
                    end=right;
                }
                left--;
                right++;
            }
                int l=i;int r=i+1;//even length
                while(l>=0 && r<s.length() && s.charAt(l)==s.charAt(r))
                {
                    if ((r-l)>(end-start))
                    {
                        start=l;
                        end=r;
                    }
                l--;
                r++;
            }
       }
       return s.substring(start,end+1);
    }
       
}